opennlp-tools/src/main/java/opennlp/tools/sentdetect/SentenceDetectorME.java

```diff
@@ -283,8 +289,15 @@ public Span[] sentPosDetect(String s) {
       }
       else {
         sentProbs.remove(si);
+        spansToRemove.add(si);
+      }
+    }
+    
+    if (spansToRemove.size() > 0) {
```

Comment:
Do we need this check? If the list is empty, it will not do anything anyway

Reply:
I was afraid of getting an out-of-index exception. So, if the list is empty, it will not even enter in the for loop? If so, I'd delete that check as you say, it'll look cleaner.

Reply:
Yes, thats's right. I the collection is empty, the for loop will not get executed.